lib/router/handler-info/unresolved-handler-info-by-object.js

```diff
@@ -34,8 +35,8 @@ var UnresolvedHandlerInfoByObject = subclass(HandlerInfo, {
     }
 
     // Use custom serialize if it exists.
-    if (handler.serialize) {
-      return handler.serialize(model, names);
+    if (serializer) {
+      return serializer(model, names);
```

Comment:
You're correct that this is a change in semantics. (Obviously. And you don't need me to agree with you to know that. 😄)

It's possible to "fix" in that we could [save off context here](https://github.com/tildeio/router.js/pull/182/files/d12ae9e58062fc7effe1557e13e704b3d58bf838#diff-fda34cdb7e594e6019ca204f39a9d00aR29) and then use `.call`. However with the route-serializers RFC that may be a bit surprising to people.

I'm personally of the opinion that I want serializers to be pure transforms, not based upon instance state. Given that, I would argue that having access to `this` was a bug itself, not a feature. Still, can you describe your use case so we can make a more informed decision?

(Also, congratulations! You may be the only person using this pattern in the entire Ember community considering the lag between release and report.)